src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -1,23 +1,32 @@
 package core.basesyntax;
 
-/**
- * <p>Реалізуйте метод `getUniqueCharacters(String fileName)` який приймає як параметр назву файлу.
- * Для цього використовуйте TreeSet. Файл містить букви латинського алфавіту і розділові знаки.</p>
- *
- * <p>Метод повинен відсортувати всі букви по алфавіту і повернути строку з 5 різних букв без
- * розділових знаків.
- * Якщо у файлі міститься менше п'яти різних букв, то потрібно повернути їх усі.
- * Можливий випадок, коли переданого файлу не існує, в такому разі потрібно викинути помилку
- * про відсутність файлу.</p>
- *
- * <p>Приклад 1: ur-BvT?^ ra w; p
- * Результат 1: abprt</p>
- *
- * <p>Приклад 2: A _f*c a?F
- * Результат 2: acf</p>
- */
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.TreeSet;
+
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws FileNotFoundException {
+        StringBuilder stringBuilder = new StringBuilder();
+        StringBuilder chapter = new StringBuilder();
+        TreeSet<Character> treeSet = new TreeSet<>();
+        try {
+            Files.lines(Paths.get(fileName)).forEach(stringBuilder::append);
+        } catch (IOException e) {
+            throw new FileNotFoundException(e.toString());
+        }
+        char[] chars = stringBuilder.toString().toCharArray();
+        for (Character i : chars) {
+            if (Character.isLetter(i)) {
+                treeSet.add(Character.toLowerCase(i));
+            }
+        }
+        int x = Math.min(treeSet.size(), 5);
+        for (int j = 0; j < x; j++) {
+            chapter.append(treeSet.first());
+            treeSet.remove(treeSet.first());
```

Comment:
u can just use for each loop and break it when it's 5

Reply:
> u can just use for each loop and break it when it's 5

Could you please explain, why it will be more corrected?